CMakeLists.txt

```
@@ -378,13 +378,19 @@ set(IREE_CUDA_LIBDEVICE_PATH "" CACHE STRING "Absolute path to an appropriate li
 if(IREE_TARGET_BACKEND_CUDA OR IREE_HAL_DRIVER_CUDA)
   find_package(CUDAToolkit)
 
-  # We define the magic IREE_CUDA_DEPS_DIR env var in our CI docker images if we
-  # have a stripped down CUDA toolkit suitable for compiling available. We
-  # trigger on this below as a fallback for locating headers and libdevice
-  # files.
   if(NOT CUDAToolkit_FOUND AND DEFINED ENV{IREE_CUDA_DEPS_DIR})
+    # We define the magic IREE_CUDA_DEPS_DIR env var in our CI docker images if we
+    # have a stripped down CUDA toolkit suitable for compiling available. We
+    # trigger on this below as a fallback for locating headers and libdevice
+    # files.
     set(CUDAToolkit_ROOT "$ENV{IREE_CUDA_DEPS_DIR}")
     message(STATUS "CUDA SDK not found by CMake but using IREE_CUDA_DEPS = ${CUDAToolkit_ROOT}")
+  else()
+    # If we haven't found CUDA deps, emulate the CI by downloading them fon demand.
```

I wouldn't really say we're "emulating the CI" here. We're just making Cuda buildable.

Also, nit: typo

```suggestion
    # If we haven't found CUDA deps, download at least enough to build for Cuda.
```